src/main/kotlin/org/rust/lang/core/types/RustType.kt

```diff
@@ -19,6 +19,13 @@ interface RustType {
         RsImplIndex.findMethodsFor(this, project)
 
     /**
+     * Checks if `other` type may be represented as this type.
+     *
+     * Note that `t1.isAssignableFrom(t2)` is not the same as `t2.isAssignableFrom(t1)`.
+     */
+    fun isAssignableFrom(other: RustType, project: Project): Boolean
```

Comment:
This is not the best name. In Java, `isAssignableFrom` is about subtyping, but there's no subtypnig in Rust (except for the lifetimes). I suggest `canUnifyWith` instead: a type parameter `T` can unify with some concrete type `S` iff `S` satisfies all the bounds imposed on `T`.